src/aws-cpp-sdk-core/include/aws/core/client/AWSErrorMarshaller.h

```diff
@@ -133,9 +133,13 @@ namespace Aws
           AWSError<CoreErrors> BuildAWSError(const std::shared_ptr<Http::HttpResponse>& httpResponse) const override;
 
         protected:
-          const Aws::Utils::Cbor::CborValue& GetCborPayloadFromError(const AWSError<CoreErrors>&) const;
           static Aws::Utils::Cbor::CborValue GetCborPayloadHttpResponse(const Http::HttpResponse& httpResponse);
         };
 
+        class AWS_CORE_API RpcV2ErrorMarshallerQueryCompatible : public RpcV2ErrorMarshaller {
```

Comment:
We need tests for this in [`AWSErrorMashallerTest`](https://github.com/aws/aws-sdk-cpp/blob/main/tests/aws-cpp-sdk-core-tests/aws/client/AWSErrorMashallerTest.cpp), notably specifically testing query compatibility